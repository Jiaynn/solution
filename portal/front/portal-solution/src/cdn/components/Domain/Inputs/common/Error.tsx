/*
 * @file component Error
 * @author nighca <nighca@live.cn>
 */

import React, { CSSProperties, ReactNode } from 'react'

import { ValidatorError, isEmpty } from 'cdn/transforms/form'

const defaultStyle: CSSProperties = {
  color: '#e55c5c',
  display: 'block',
  verticalAlign: 'middle',
  marginTop: '8px',
  lineHeight: '20px'
}

export interface Props {
  style?: CSSProperties
  error?: ValidatorError
  children?: ReactNode
}

export default function Error(props: Props) {
  if (isEmpty(props.error) && !props.children) {
    return null
  }

  return (
    <span
      className="error-wrapper"
      style={{ ...defaultStyle, ...props.style }}
    >
      {props.children || props.error}
    </span>
  )
}
