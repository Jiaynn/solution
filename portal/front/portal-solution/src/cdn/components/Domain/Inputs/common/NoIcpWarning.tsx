/**
 * @file NoIcpWarning Component
 * @author linchen <gakiclin@gmail.com>
 */

import React from 'react'

export default function NoIcpWarning(props: { desc?: React.ReactNode}) {
  return (
    <div className="comp-no-icp-warning">
      <span className="icp-text">域名未备案，</span>
      <span className="icp-desc">{props.desc || '覆盖范围只能选择海外'}</span>
    </div>
  )
}
