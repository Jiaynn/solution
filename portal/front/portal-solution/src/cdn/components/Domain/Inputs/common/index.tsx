import React from 'react'

// 给传入的列表，每两项间添加一个带 key 的 <hr />
export function addHr(items: JSX.Element[]) {
  return items.filter(
    item => item != null
  ).reduce((current, item, index) => {
    if (index > 0) {
      current.push(<hr className="sep-line" key={`hr-${index}`} />)
    }
    current.push(item)
    return current
  }, [] as JSX.Element[])
}
