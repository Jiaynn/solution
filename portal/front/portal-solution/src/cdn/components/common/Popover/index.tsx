/**
 * @file component Popover
 * @description 指定了 getPopupContainer 的 Popover，升级 icecream2 后不需要用该组件，它已经内置该功能
 * @author liaoxiaoyou <liaoxiaoyou@qiniu.com>
 */

import React, { useContext } from 'react'
import IcecreamPopover, { PopoverProps } from 'react-icecream/lib/popover'
import { PopupContainerContext } from 'react-icecream/lib/popup-container'

export { PopoverProps } from 'react-icecream/lib/popover'

export default function Popover(props: PopoverProps) {
  const containerContext = useContext(PopupContainerContext)

  return (
    <IcecreamPopover
      getPopupContainer={containerContext.getContainer}
      {...props}
    />
  )
}
