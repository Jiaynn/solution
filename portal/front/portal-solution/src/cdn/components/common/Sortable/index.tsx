/**
 * @file Sortable Component
 * @author linchen <gakiclin@gmail.com>
 */

import React from 'react'
import { SortableHandle } from 'react-sortable-hoc'

import MoveSvg from './images/move.svg'

import './style.less'

export { SortableElement, SortableContainer, arrayMove, SortEnd } from 'react-sortable-hoc'

export const SortDragHandle = SortableHandle(() => (
  <div className="comp-handle-move"><MoveSvg className="icon-move" /></div>
))
