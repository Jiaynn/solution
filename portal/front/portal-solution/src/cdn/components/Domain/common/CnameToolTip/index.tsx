/**
 * @file CnameToolTip Component
 * @author linchen gakiclin@gmail.com
 */

import React from 'react'
import Icon from 'react-icecream/lib/icon'
import Tooltip from 'react-icecream/lib/tooltip'
import { useTranslation } from 'portal-base/common/i18n'

import cnameDoc from 'cdn/docs/config-cname.pdf'

import HelpLink from 'cdn/components/common/HelpLink'

import './style.less'

const cnameTipMsg = {
  cn: (
    <>
      如需使用 CDN 加速，需要配置加速域名指向 CNAME。配置 CNAME 后大约有 10 分钟延迟才会生效。
      <HelpLink oemHref={cnameDoc} href="https://developer.qiniu.com/fusion/kb/1322/how-to-configure-cname-domain-name">如何配置</HelpLink>
    </>
  ),
  en: (
    <>
      To use CDN acceleration, you need to configure the accelerated domain name to point to CNAME.
      After CNAME is configured, it will take about 10 minutes to take effect.
    </>
  )
}

export default function CnameToolTip() {
  const t = useTranslation()

  const title = (
    <div style={{ fontSize: '12px' }}>
      {t(cnameTipMsg)}
    </div>
  )

  return (
    <span className="comp-cname-tool-tip">
      CNAME
      <Tooltip title={title}>
        <Icon className="tool-tip-icon" type="question-circle" theme="filled" />
      </Tooltip>
    </span>
  )
}
