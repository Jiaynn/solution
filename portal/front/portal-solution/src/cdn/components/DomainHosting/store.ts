/**
 * @file 域名托管 store
 * @author linchen <linchen@qiniu.com>
 */

import autobind from 'autobind-decorator'
import { observable, computed, action } from 'mobx'
import { ToasterStore } from 'portal-base/common/toaster'
import { Loadings } from 'portal-base/common/loading'
import Store, { observeInjectable as injectable } from 'qn-fe-core/store'

import { ModalStore } from 'cdn/stores/modal'

import { DomainStatus } from 'cdn/constants/domain-hosting'

import DomainHostingApis, { IDomainInfo, IGetDomainListResp } from 'cdn/apis/oem/domain-hosting'
import { IExtraProps as IDomainModalExtraProps, IValue as IDomainModalValue } from './Modal/index'

enum LoadingType {
  GetDomainList = 'GetDomainList',
  CreateDomain = 'CreateDomain',
  UpdateDomain = 'UpdateDomain',
  CheckDomainStatus = 'CheckDomainStatus'
}

@injectable()
export default class LocalStore extends Store {
  loading = Loadings.collectFrom(this, LoadingType)
  domainModalStore = new ModalStore<IDomainModalExtraProps, IDomainModalValue>()

  @observable.shallow domainList: IDomainInfo[] = []

  constructor(
    private domainHostingApis: DomainHostingApis
  ) {
    super()
  }

  @computed get isLoading() {
    return this.loading.isLoading(LoadingType.GetDomainList)
  }

  @computed get domainExists() {
    return this.domainList && this.domainList.length > 0
  }

  @computed get domainStatus() {
    return this.domainExists ? this.domainList[0].status : DomainStatus.Invalid
  }

  @action.bound updateDomainList(resp: IGetDomainListResp) {
    this.domainList = resp.records
  }

  @action.bound handleUpdateDomainDone(item: IDomainInfo) {
    const index = this.domainList.findIndex(it => it.id === item.id)
    if (index !== -1) {
      this.domainList.splice(index, 1, item)
    }
  }

  @autobind
  @ToasterStore.handle()
  @Loadings.handle(LoadingType.CheckDomainStatus)
  checkDomain(domain: string) {
    return this.domainHostingApis.checkDomain(domain).then(this.getDomainList)
  }

  @autobind
  @ToasterStore.handle()
  @Loadings.handle(LoadingType.GetDomainList)
  getDomainList() {
    return this.domainHostingApis.getDomainList().then(this.updateDomainList)
  }

  @autobind
  @ToasterStore.handle()
  @Loadings.handle(LoadingType.CreateDomain)
  createDomain(domain: string) {
    return this.domainHostingApis.createDomain(domain).then(this.getDomainList)
  }

  @autobind
  @ToasterStore.handle()
  @Loadings.handle(LoadingType.UpdateDomain)
  updateDomain(item: IDomainInfo) {
    return this.domainHostingApis.updateDomain(item).then(() => this.handleUpdateDomainDone(item))
  }
}
