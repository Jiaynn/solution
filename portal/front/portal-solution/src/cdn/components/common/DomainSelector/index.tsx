/**
 * @file DomainSelector Component
 * @author linchen <gakiclin@gmail.com>
 */

import React, { useCallback, ReactNode } from 'react'
import { observer } from 'mobx-react'
import Input from 'react-icecream/lib/input'
import Select from 'react-icecream/lib/select'
import { useInjection } from 'qn-fe-core/di'
import { I18nStore } from 'portal-base/common/i18n'
import { useLocalStore } from 'portal-base/common/utils/store'

import { MAX_DOMAIN_COUNT } from 'cdn/constants/domain'

import { SimpleTagSelector } from 'cdn/components/common/TagSelector'
import Popover, { PopoverProps } from 'cdn/components/common/Popover'

import { IQueryParams } from 'cdn/apis/domain'

import DomainFilter from './DomainFilter'
import ProtocolFilter from './ProtocolFilter'
import DomainList from './DomainList'
import FullSelector from './FullSelector'

import LocalStore, { State } from './store'

import './style.less'

export { createState, getValue, State } from './store'

export interface IProps {
  withTags?: boolean
  showFullCheck?: boolean
  queryParams?: Partial<IQueryParams>
  state: State
}

export default observer(function DomainSelector(props: IProps) {
  const store = useLocalStore(LocalStore, props)

  const selectorCnt = (
    <>
      <DomainFilter
        state={store.props.state.$.domainFilter}
        onSearch={store.searchDomains}
      />
      <ProtocolFilter
        state={store.props.state.$.protocolFilter}
      />
      <DomainList
        loading={store.isLoading}
        domainList={store.domainsForSelect}
        state={store.props.state.$.domains}
      />
      <FullSelector
        total={store.total}
        maxDomainCount={MAX_DOMAIN_COUNT}
        state={store.props.state.$.fullSelector}
        showFullCheck={props.showFullCheck}
      />
    </>
  )

  const handleVisibleChange = useCallback((visible: boolean) => {
    if (visible) {
      store.reorderDomains()
    }
  }, [store])

  return (
    <div className="comp-domain-selector">
      <Input.Group compact>
        {
          store.shouldShowTags && (
            <SimpleTagSelector
              style={{ marginRight: '8px' }}
              state={store.props.state.$.tags}
            />
          )
        }
        <DomainSelectorContainer
          content={selectorCnt}
          onVisibleChange={handleVisibleChange}
        >
          <DomainSelectorSummary
            isLoading={store.isLoading}
            isFullDomainChecked={store.isFullDomainChecked}
            selectedTagCount={store.selectedTags.length}
            selectedDomainCount={store.selectedDomains.length}
          />
        </DomainSelectorContainer>
      </Input.Group>
    </div>
  )
})

interface DomainSelectorSummaryProps {
  isLoading: boolean
  selectedTagCount: number
  selectedDomainCount: number
  isFullDomainChecked: boolean
}

function DomainSelectorSummary({
  isLoading,
  selectedTagCount,
  selectedDomainCount,
  isFullDomainChecked
}: DomainSelectorSummaryProps) {
  const i18n = useInjection(I18nStore)
  return (
    <Select
      loading={isLoading}
      style={{ width: 200 }}
      value={getSummary(i18n, selectedTagCount, selectedDomainCount, isFullDomainChecked)}
      dropdownRender={() => <span></span>}
    />
  )
}

const messages = {
  fulldomainChecked: {
    cn: '已选择全量域名',
    en: 'All domain names selected'
  },
  domainSelected: {
    cn: (num: number) => `已选 ${num} 个域名`,
    en: (num: number) => `${num} domain names selected.`
  },
  pleaseSelectDomain: {
    cn: '请选择域名',
    en: 'Please select a domain name'
  }
}

function getSummary(
  i18n: I18nStore,
  selectedTagCount: number,
  selectedDomainCount: number,
  isFullDomainChecked: boolean
) {
  if (isFullDomainChecked) {
    return i18n.t(messages.fulldomainChecked)
  }
  if (selectedDomainCount > 0) {
    return i18n.t(messages.domainSelected, selectedDomainCount)
  }
  if (selectedTagCount > 0) {
    return i18n.t(messages.domainSelected, 0)
  }
  return i18n.t(messages.pleaseSelectDomain)
}

interface DomainSelectorContainerProps extends Pick<PopoverProps, 'onVisibleChange'> {
  content: ReactNode
  children: ReactNode
}

function DomainSelectorContainer({
  content,
  children,
  onVisibleChange
}: DomainSelectorContainerProps) {
  return (
    <div className="comp-domain-selector">
      <Popover
        trigger="click"
        placement="bottomLeft"
        onVisibleChange={onVisibleChange}
        overlayClassName="domain-selector-overlay"
        content={<div className="domain-selector">{content}</div>}
        autoAdjustOverflow={false}
      >
        <div>
          {children}
        </div>
      </Popover>
    </div>
  )
}
