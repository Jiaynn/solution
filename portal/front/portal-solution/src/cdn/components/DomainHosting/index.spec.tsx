import React from 'react'
import { RouterStore } from 'portal-base/common/router'
import { LocalStorageStore } from 'portal-base/common/utils/storage'

import { createOemRendererWithRouter } from 'test'

import { localStorageKey } from 'cdn/constants/domain-hosting'
import DomainHosting from '.'

const renderer = createOemRendererWithRouter()

beforeAll(() => {
  const storageStore = renderer.inject(LocalStorageStore)
  storageStore.setItem(localStorageKey, true)
})

it('renders correctly', () => {
  const routerStore = renderer.inject(RouterStore)
  routerStore.push('/domain-hosting')
  const tree = renderer.createWithAct(
    <DomainHosting />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
