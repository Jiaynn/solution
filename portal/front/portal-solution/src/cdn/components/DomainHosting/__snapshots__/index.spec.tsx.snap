// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="comp-domain-hosting"
>
  <div
    className="icecream-breadcrumb"
  />
  <div
    className=""
  >
    <div
      className="domain-hosting-header"
    >
      <button
        className="ant-btn"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          aria-label="icon: plus"
          className="anticon anticon-plus"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </i>
        <span>
          添加托管域名
        </span>
      </button>
      <i
        aria-label="icon: info-circle"
        className="anticon anticon-info-circle add-tip"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="info-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </i>
    </div>
    <div
      className="ant-table-wrapper domain-hosting-list"
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div>
          <div
            className="ant-spin ant-spin-spinning ant-table-without-pagination ant-table-spin-holder"
          >
            <span
              className="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
        <div
          className="ant-spin-container ant-spin-blur"
        >
          <div
            className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
            style={Object {}}
          >
            <div
              className="ant-table-content"
            >
              <div
                className="ant-table-body"
                onScroll={[Function]}
                onWheel={[Function]}
                style={Object {}}
              >
                <table
                  className=""
                  style={Object {}}
                >
                  <colgroup>
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr
                      style={
                        Object {
                          "height": null,
                        }
                      }
                    >
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              域名
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              状态
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              操作
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  />
                </table>
              </div>
              <div
                className="ant-table-placeholder"
              >
                <div
                  className="ant-empty ant-empty-normal"
                >
                  <div
                    className="ant-empty-image"
                  >
                    <img
                      alt="No Data"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                    />
                  </div>
                  <p
                    className="ant-empty-description"
                  >
                    No Data
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-spin-nested-loading"
    >
      <div>
        <div
          className="ant-spin ant-spin-spinning"
        >
          <span
            className="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        className="ant-spin-container ant-spin-blur"
      >
        <div
          className="domain-hosting-operation-tip"
        >
          <p
            className="domain-hosting-operation-tip-title"
          >
            域名托管操作索引：
          </p>
          <p
            className="domain-hosting-operation-tip-item"
          >
            场景一：若您的域名已托管在 DNSPod 平台，请将该域名的管理权授权给 ID：100005111910
          </p>
          <p
            className="domain-hosting-operation-tip-item"
          >
            场景二：若您的域名未添加至 DNSPod 请到您的域名购买处修改 DNS 服务器，并指向如下两个地址：f1g1ns1.dnspod.net、 f1g1ns2.dnspod.net
          </p>
          <p
            className="domain-hosting-operation-tip-remark"
          >
            注：配置一旦生效，请务必保持该授权持续有效，否则会影响到您客户的解析服务。
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
